libpod/healthcheck.go

```diff
@@ -54,12 +55,21 @@ func (r *Runtime) HealthCheck(ctx context.Context, name string) (define.HealthCh
 	return hcStatus, err
 }
 
-func (c *Container) runHealthCheck(ctx context.Context, isStartup bool) (define.HealthCheckStatus, string, error) {
+func (c *Container) runHealthCheck(ctx context.Context, isStartup bool) (returnedStatus define.HealthCheckStatus, _ string, _ error) {
 	var (
 		newCommand    []string
 		returnCode    int
 		inStartPeriod bool
 	)
+
+	// Write HC event with appropriate status after we return
+	defer func() {
+		if returnedStatus == define.HealthCheckNotDefined || returnedStatus == define.HealthCheckInternalError {
+			return
+		}
+		c.newContainerEvent(events.HealthStatus)
+	}()
```

Comment:
Why is this defer? Putting it directly before the return should be enough and does not require the unnecessary if statement.

Reply:
@mheon PTAL

Reply:
Mostly a hedge against the addition of further return statements to the function, could definitely be changed to avoid a defer

Reply:
I don't see how this would make a difference, the new return will be added either before in which case we do not want to send health status anyway or afterwards in which case the event was already written which should be fine.

I find that much better as this does not need the extra if HealthCheckNotDefined check

Reply:
I am OK both ways.  Let's get this PR in. We slipped the 4.6.1 release.